mitodl/mit-learn
# Card Accessibility Improvements

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5538

### Description  (What does it do?)
This PR makes some accessibility enhancements to cards and learning resource cards in particular. 

**Card Title is Link:** Previously the entire card was a link. Now just the title is a link, leading to much better link names. *The whole card remains clickable.*

**LearningResourceCards are `article`s**: Learning resource cards now use `article` as a root element with `[resource.type]: [resource.title]` as the article name. Consequently, cards are landmarks that appear in landmark navigation

**Simplified Positioning**: Cards no longer use absolute positioning for buttons. 

### Screenshots

![landmarks_rotor](https://github.com/user-attachments/assets/22a6a3de-be46-4023-8f89-e58a3c583798)
![links_rotor](https://github.com/user-attachments/assets/9fc9ae27-e56f-4e0b-b601-0791eecd1c80)

Card aesthetics should be unchanged:


<img width="1336" alt="Screenshot 2024-10-30 at 7 57 45 AM" src="https://github.com/user-attachments/assets/602c5491-f81d-46f0-a21c-5ba080abe588">
<img width="1403" alt="Screenshot 2024-10-30 at 7 57 55 AM" src="https://github.com/user-attachments/assets/ad242370-4d87-4d9e-a2d5-7dadde20f197">
<img width="1119" alt="Screenshot 2024-10-30 at 7 58 04 AM" src="https://github.com/user-attachments/assets/2b942839-3424-4ec0-a994-9abb750dc02c">
<img width="1095" alt="Screenshot 2024-10-30 at 7 58 12 AM" src="https://github.com/user-attachments/assets/efbd183b-9dad-4164-a4f1-e5f6c976ee9f">

### How can this be tested?
1. Inspect cards and check that their aesthetic is unchanged. Cards include:
    - Carousel resource cards (LearningResourceCard)
    - Search result cards (LearningResourceListCard)
    - Userlist cards at `/dashboard/my-lists/` 
    - Draggable resource cards at `/dashboard/my-lists/{list_id}` when "Reordering"
    - News & Events cards on the homepage 
2.Where cards were clickable as links before, they should be now.
    - You should still be able to ctrl+click to open in a new tab
3. Interactive children in cards should still work, and should NOT navigate the whole link.  Examples are
   - "AddToListDialog" (learning path and userlist versions) 
   - The "..." menu on learning path cards.
